File: src/main/kotlin/com/nano/MathController.kt
package com.nano

import io.micronaut.http.HttpResponse
import io.micronaut.http.MediaType
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.Post
import io.micronaut.validation.Validated

@Controller("/calculator")
@Validated
class MathController {

     @Get("/greeting")
    fun greeting(): HttpResponse<String> {
        return HttpResponse.ok("Hello world!")
    }

    @Post("/add", produces = [MediaType.APPLICATION_JSON])
    fun add(first: Int, second: Int): HttpResponse<Map<String, Int>> {
        val result = first + second
        return HttpResponse.ok(mapOf("result" to result))
    }

    @Post("/subtract", produces = [MediaType.APPLICATION_JSON])
    fun subtract(first: Int, second: Int): HttpResponse<Map<String, Int>> {
        val result = first - second
        return HttpResponse.ok(mapOf("result" to result))
    }
}

File: MathController.py
from flask import Flask, request, jsonify
from flask.wrappers import Response
from dataclasses import dataclass

app = Flask(__name__)

@dataclass
class Numbers:
    first: int
    second: int

@app.route('/calculator/greeting', methods=['GET'])
def greeting():
    return "Hello World!"

@app.route('/calculator/add', methods=['POST'])
def add():
    numbers = Numbers(**request.json)
    result = numbers.first + numbers.second
    return jsonify({"result": result})

@app.route('/calculator/subtract', methods=['POST'])
def subtract():
    numbers = Numbers(**request.json)
    result = numbers.first - numbers.second
    return jsonify({"result": result})

if __name__ == '__main__':
    app.run(debug=True)
